File: Avaliacoes/CSharp/Modulo2/thread/FilialThread.cs
using System.Threading;

/// <summary>
/// Thread responsável por calcular o faturamento de UMA filial, de forma
/// totalmente isolada — sem acessar nenhuma variável global ou
/// compartilhada com as outras threads.
/// </summary>
public class FilialThread
{
    private readonly Thread _thread;
    private readonly Filial _filial;
    private readonly ResultadoFilial _resultado = new ResultadoFilial();
    private readonly RelatorioView _view;

    public string Nome { get; }

    public FilialThread(Filial filial, RelatorioView view)
    {
        Nome = $"Filial-{filial.Id}";
        _filial = filial;
        _view = view;
        _thread = new Thread(Executar) { Name = Nome };
    }

    /// <summary>
    /// Soma o faturamento em uma variável LOCAL (não estática, não
    /// compartilhada) e guarda o resultado na própria instância de
    /// <see cref="ResultadoFilial"/>, que só esta thread escreve.
    /// </summary>
    private void Executar()
    {
        long somaLocal = 0;
        foreach (int valor in _filial.Vendas)
        {
            somaLocal += valor;
        }

        _resultado.IdFilial = _filial.Id;
        _resultado.FaturamentoTotal = somaLocal;

        _view.ExibirResultadoParcial(_resultado);
    }

    public void Start() => _thread.Start();
    public void Join() => _thread.Join();

    /// <summary>
    /// Retorna o resultado calculado por esta thread. Só deve ser chamado
    /// pelo Controller DEPOIS do <see cref="Join"/>, quando não há mais
    /// concorrência sobre o objeto.
    /// </summary>
    public ResultadoFilial GetResultado() => _resultado;

    public Filial GetFilial() => _filial;
}

File: Avaliacoes/Java/Modulo2/src/main/java/thread/FilialThread.java
package thread;

import model.Filial;
import model.ResultadoFilial;
import view.RelatorioView;

/**
 * Thread responsável por calcular o faturamento de UMA filial, de forma
 * totalmente isolada — sem acessar nenhuma variável global ou
 * compartilhada com as outras threads.
 *
 * @author Andrisa Santos
 */
public class FilialThread extends Thread {

    /** Dados de entrada exclusivos desta thread (não compartilhados). */
    private final Filial filial;

    /** Resultado exclusivo desta thread (só ela escreve nele). */
    private final ResultadoFilial resultado;

    private final RelatorioView view;

    public FilialThread(Filial filial, RelatorioView view) {
        super("Filial-" + filial.getId());
        this.filial = filial;
        this.resultado = new ResultadoFilial();
        this.view = view;
    }

    /**
     * Regra do exercício: em nenhum momento desta execução esta thread
     * pode ler ou escrever em uma variável global/estática compartilhada
     * com as outras threads.
     */
    @Override
    public void run() {
        long somaLocal = 0;
        for (int valor : filial.getVendas()) {
            somaLocal += valor;
        }

        resultado.setIdFilial(filial.getId());
        resultado.setFaturamentoTotal(somaLocal);

        view.exibirResultadoParcial(resultado);
    }

    /**
     * Retorna o resultado calculado por esta thread.
     * Só deve ser chamado pelo Controller DEPOIS do {@code join()} desta
     * thread — momento em que não há mais concorrência sobre o objeto.
     *
     * @return resultado do cálculo desta filial
     */
    public ResultadoFilial getResultado() {
        return resultado;
    }

    /** @return os dados de entrada (filial) associados a esta thread */
    public Filial getFilial() {
        return filial;
    }
}
